### ISSUE TITLE
[nn] Passing dtype to `_stacklevel` argument in `log_softmax` silently does nothing

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#61574]
Docs: https://pytorch.org/docs/stable/generated/torch.nn.functional.log_softmax.html

Signature: `torch.nn.functional.log_softmax(input, dim=None, _stacklevel=3, dtype=None)`

Passing `torch.dtype` to `_stacklevel` should ideally raise an error but it works silently discarding the dtype.

```python
>>> import torch
>>> t = torch.ones(2,2)
>>> torch.nn.functional.log_softmax(t, 0, torch.float64)
tensor([[-0.6931, -0.6931],
        [-0.6931, -0.6931]]) # Not torch.float64
```

`dtype` should have earlier position than `_stacklevel`. Also ideally `_stacklevel` should be a kwarg-only argument. But those changes would be BC-breaking.

**Comment by ngimel** [comment]
I've looked at torch.nn.functional.log_softmax usages, and haven't found any that would specify more than 2 arguments, so I think we can do a bc-breaking change here.

**Comment by vadimkantorov** [comment]
I also wonder why only log_softmax has _stacklevel as argument...
